src/hashing.rs

```diff
@@ -303,43 +304,65 @@ pub trait AsFixedBytes<const I: usize>: AsRef<[u8]> {
         let hash_vec = self.as_ref();
         if hash_vec.is_empty() || hash_vec.len() < I {
             let hash_vec_length = if hash_vec.is_empty() { 0 } else { hash_vec.len() };
-            return Err(SliceError::CopyFromSlice(I, hash_vec_length));
+            return Err(SliceError::CopyFromSlice {
+                target: I,
+                provided: hash_vec_length,
+            });
         }
         let mut buffer: [u8; I] = [0; I];
         buffer.copy_from_slice(&hash_vec[..I]);
         Ok(buffer)
     }
 }
 
+impl<TInnerDigest: OutputSizeUser, TDomain: DomainSeparation> OutputSizeUser
```

Comment:
These changes seem extraneous to the PR and should be put in a separate PR. It's hard to know what the purpose of these changes are